### ISSUE TITLE
Disable update notification

### ISSUE THREAD

**Issue by sausje** [raiguard/ModernGadgets#109]
I really dislike being harassed to update, with the only option to skip the update is unloading the skin..
I've now blocked ModernGadgets from contacting the update server, but this is not an acceptable solution. You should really give people the choice to opt out of updates.
Especially since every update resets my settings and can't seem to put the backup back.

**Comment by raiguard** [comment]
The update notifications are not forced. You can turn them off through the global settings skin. There's also a "never show this again" button at the bottom of the notification. I'm not forcing it on anybody. If you really want to be sure they're gone, unload the setup skin (the one that shows the notification).

The backup import works for 99% of people. I'd need more information if I were to diagnose your problem.

**Comment by raiguard** [comment]
Yeah, the button disables the notifications then unloads the skin. Theoretically, your should never see them again unless you re-enable them through the global settings.

If you upgraded from 1.3.1 to 1.4, your HWiNFO settings won't have carried over because I had to change all the variable names. Is that what you're referring to?

**Comment by sausje** [comment]
See my edit for the confusion in defining the unload skin part, maybe better to change it? Or make it more clear that it's just that part of the skin that unloads?

No what I referred to was after the update that brought change to how you process the values. (Although IMO the old method was just fine too.)

**Comment by raiguard** [comment]
So yes, it's from 1.3.1 to 1.4.

I just had a brain wave: if you unloaded the notification skin then your settings won't be backed up either. When you updated, did you do it because of the notification or of your own volition?

**Comment by raiguard** [comment]
Can you summarize what customizations you had made, and what wasn't reverted when you imported the backup?

**Comment by sausje** [comment]
Just on the disk meter, the hidden disks didn't set themselves back and the temps started showing again even though I disabled it (because of some inaccurate temps when having multiple partitions).

And to come back, no, all I mentioned was after the big 1.4 update. It had (I think) 2-3 smaller updates after that, which I referred to.
The 1.4 update I applied myself, but everything after that was with auto-update.

**Comment by raiguard** [comment]
Huh that's odd. You must have been on development versions because disk temps didn't exist until 1.4. I'll try and see what I can find out tomorrow.

**Comment by raiguard** [comment]
Hey, can you enable development updates, install the v1.5 beta, and let me know if everything works? It should automatically preserve your settings now, rather than showing a prompt.

**Comment by raiguard** [comment]
I'm gonna close this due to lack of response, and hopefully the problem was fixed with the new system anyway.

**Comment by sausje** [comment]
Apologies, been quite busy.

I just updated to beta 2 and it reset EVERYTHING after I even specifically told it to make a backup of my settings.
Even the layout placement is gone, unloaded and everything.

**Comment by raiguard** [comment]
What the... You mean it reset your settings as well? The layout thing was expected, I changed the names of all of the config files to make more sense so all of the existing skins will unload.

But did it lose ALL of your settings as well? If so then that speaks to something being wrong with your Rainmeter installation itself.

**Comment by sausje** [comment]
Yeah oddly enough the first time I pressed the download button, it downloaded RC1 version again and that reset my settings.
I've now done a fully clean setup of the skin, so we'll see on the next update if settings this time stick or not.

**Comment by raiguard** [comment]
If you're willing, you can actually test it without waiting for another update. Click [here](https://github.com/raiguard/ModernGadgets/releases/download/v1.5.0-beta.2/ModernGadgets_1.5.0-beta.2.rmskin) to download the v1.5.0-beta.2 file, then open it to install it. It SHOULD keep your settings (and this time it won't unload all of the gadgets). Could you let me know if it works or not?

**Comment by sausje** [comment]
The only thing it didn't remember from my settings after installing that one was my filtered disks. Other than that it's good now.

**Comment by raiguard** [comment]
That's odd, it should still remember the filtered disks. I'll look into it. Thanks for testing, and sorry for all of the inconveniences.

**Comment by Shrekmachine** [comment]
I can confirm that behaviour with my installation.

**Comment by raiguard** [comment]
O.O

Can you reinstall again while viewing the Disks settings file? You can open it by clicking the button next to the close button in the disks meter settings skin. Tell me the value of `manualHideDisks` before the reinstallation, then what it is after reinstallation (you'll probably have to close and open the file again to see the changes).

I would test this myself but I am away from home at the moment and would like to get as much information as possible so I can fix it faster.

**Comment by sausje** [comment]
It leaves the values empty after: https://i.imgur.com/XDyuWUy.png (before left, after right)

**Comment by raiguard** [comment]
Try adding quotes around the letters `"H I"` and try again.

**Comment by raiguard** [comment]
Well, I'm home now, and that didn't work either. As far as I can tell, that option is the ONLY one in the skin that behaves like this. All of the other options work normally...

I even tried it with all Disks skins unloaded and it still did it. This is very strange...

**Comment by raiguard** [comment]
Well I figured out the problem... it's not one that I can fix myself! It's a problem with Rainmeter...

Turns out that if you have a blank value as the default, after updating, the value will always be overwritten with the blank, rather than keeping the previously set value.

Here's an example:

Current settings file:
```
[Variables]
manualHideDisks=D E
foo=The fox says OWWWWWWW
```

New settings file:
```
[Variables]
manualHideDisks=
foo=bar
```

Resulting settings file:
```
[Variables]
manualHideDisks=
foo=The fox says OWWWWWWW
```

**Comment by raiguard** [comment]
So I talked to the Rainmeter developers and this issue will not be fixed anytime soon. Therefore, I'll simply have to add a workaround that avoids having any blank values. It'll require some convoluted stuff, but it'll work. I'll release the changes with beta 3, and hopefully they will fix the issue.

**Comment by raiguard** [comment]
So about the update notification... I decided to make it slightly less intrusive. It'll now appear as a banner in the bottom-right of your main monitor:

![2019-01-13 13_33_31-window](https://user-images.githubusercontent.com/3515394/51090347-de327680-1737-11e9-9369-c8204be82a6a.png)

It plays a sound, but that can be disabled.

I have also made the update check option much more prominent. Now you can disable it from the welcome screen, which is what you see when you install ModernGadgets for the first time:

![2019-01-12 23_41_22-movies tv](https://user-images.githubusercontent.com/3515394/51090340-be9b4e00-1737-11e9-8ab6-fedb08275ddc.png)

So now, if you don't want to be pestered, you can disable it right away.

I hope these changes are acceptable.

As for the carrying-over of settings, I cannot find a way to work around the bug in Rainmeter without redesigning basically everything about how the Hide Disks functionality works. So I'm just going to leave it as-is, That, along with the Network Interface options, are the only two to be affected by this issue.

**Comment by raiguard** [comment]
So I'm going to close this issue now, seeing as the discussion is complete, and the only thing to do to fix the issue is wait for the Rainmeter devs to fix it. Thanks for the feedback, and I hope you continue to enjoy my creation! :)
